lighthouse-core/gather/gatherers/dobetterweb/tags-blocking-first-paint.js

```diff
@@ -46,7 +46,7 @@ function installMediaListener() {
  */
 /* istanbul ignore next */
 function collectTagsThatBlockFirstPaint() {
-  return new Promise((resolve, reject) => {
+  return new __nativePromise((resolve, reject) => {
```

Comment:
This is really awful DX, and ripe for errors (proof is that you found 5 cases where we missed this!)

What if, in `Driver.evaluateAsync`, we did something like...

```
(() => {
  var Promise = __nativePromise;
  var Error = __nativeError;
  (${pageFunCode})();
})();
```

ie. create a scope where we define Promise, Error, etc. vars, so any code we pass in will resolve to these variables over the global objects?

(not in this PR, but I could follow up...)

wdyt?

Reply:
we already do this for `URL`, I'm not sure we should split into another PR when this is the first PR that even starts using these outside driver in the first place. 

looks like @paulirish already agrees this is a good idea, how do you feel about reverting these changes and typedefs in favor of connor's suggestion? (if they're necessary to land I guess we *can't* split, but if they were just gravy on top, revert only?)

Reply:
Patrick sorted out the native Promise stuff in #12381

> This also completes the colocation of all the cached native definition and usage (except ElementMatches), so no need to reference window.__nativePromise ever again :D